src/Command/MarkMigrated.php

```diff
@@ -150,4 +159,95 @@ protected function markAllMigrated($path)
         }
         $adapter->commitTransaction();
     }
+
+    /**
+     * Decides based on $input if it will try to mark more than one migration
+     *
+     * @param InputInterface $input
+     * @return boolean
+     */
+    protected function isBatchMarking(InputInterface $input)
+    {
+        $version = $input->getArgument('version');
+
+        return $this->isAllVersion($version) ||
+            $this->isSkipTo($input) ||
+            $this->isUpTo($input);
+    }
+
+    /**
+     * Decides based on $input which versions it should mark as migrated
+     *
+     * @param InputInterface $input
+     * @return array of versions that should be marked as migrated
+     */
+    protected function versionsToMark(InputInterface $input)
+    {
+        $version = $input->getArgument('version');
+        $migrations = $this->getManager()->getMigrations();
+        $versions = array_keys($migrations);
+
+        if ($this->isAllVersion($version)) {
+            return $versions;
+        }
+
+        $lengthIncrease = $this->isSkipTo($input) ? 0 : 1;
+        $index = array_search($version, $versions);
+
+        if (FALSE === $index) {
+            return [];
+        }
+
+        return array_slice($versions, 0, $index + $lengthIncrease);
+    }
+
+    /**
+     * Checks if the $version is for all migrations
+     *
+     * @param string $version
+     * @return boolean
+     */
+    protected function isAllVersion($version)
+    {
+        return $version == 'all' || $version == '*';
+    }
+
+    /**
+     * Checks if the $input has skip-to option
+     *
+     * @param InputInterface $input
+     * @return boolean
+     */
+    protected function isSkipTo(InputInterface $input)
+    {
+        return NULL !== $input->getOption('skip-to');
+    }
+
+    /**
+     * Checks if the $input has up-to option
+     *
+     * @param InputInterface $input
+     * @return boolean
+     */
+    protected function isUpTo(InputInterface $input)
+    {
+        return NULL !== $input->getOption('up-to');
```

Comment:
We usually do it the other way around with the null lowercased :

``if ($var === null)``